frontend/app/src/components/asset-manager/UnderlyingTokenManager.vue

```diff
@@ -101,28 +103,41 @@ const tokenKind = ref<EvmTokenKind>(EvmTokenKind.ERC20);
 const underlyingWeight = ref<string>('');
 const form = ref<any>(null);
 
-const addressRules = [
-  (v: string) =>
-    !!v || tc('underlying_token_manager.validation.address_non_empty')
-];
-
-const weightRules = [
-  (v: string) => {
-    if (!v) {
-      return tc('underlying_token_manager.validation.non_empty');
-    }
-    const number = parseInt(v);
-    if (isNaN(number)) {
-      return tc('underlying_token_manager.validation.not_valid');
-    }
-
-    if (number < 1 || number > 100) {
-      return tc('underlying_token_manager.validation.out_of_range');
-    }
-
-    return true;
+const rules = {
+  address: {
+    required: helpers.withMessage(
+      tc('underlying_token_manager.validation.address_non_empty'),
+      required
+    )
+  },
+  weight: {
+    required: helpers.withMessage(
+      tc('underlying_token_manager.validation.non_empty'),
+      required
+    ),
+    notNaN: helpers.withMessage(
+      tc('underlying_token_manager.validation.not_valid'),
+      (v: string): boolean => !isNaN(parseInt(v))
```

Comment:
Can't we use some of the default rules here for the numbers? 

Maybe something like https://vuelidate-next.netlify.app/validators.html#integer